DSharpPlus.Rest/DiscordRestClient.Channels.cs

```diff
@@ -0,0 +1,582 @@
+{
+    /// <summary>
+    /// Gets a channel object
+    /// </summary>
+    /// <param name="id">Channel ID</param>
+    public Task<DiscordChannel> GetChannelAsync(ulong id)
+        => ApiClient.GetChannelAsync(id);
+
+    /// <summary>
+    /// Gets channels from a guild
+    /// </summary>
+    /// <param name="guildId">Guild ID</param>
+    public Task<IReadOnlyList<DiscordChannel>> GetGuildChannelsAsync(ulong guildId)
+        => ApiClient.GetGuildChannelsAsync(guildId);
+
+    /// <summary>
+    /// Deletes a channel
+    /// </summary>
+    /// <param name="id">Channel ID</param>
+    /// <param name="reason">Reason why this channel was deleted</param>
+    public Task DeleteChannelAsync(ulong id, string reason)
+        => ApiClient.DeleteChannelAsync(id, reason);
+
+    /// <summary>
+    /// Creates a guild channel
+    /// </summary>
+    /// <param name="id">Channel ID</param>
+    /// <param name="name">Channel name</param>
+    /// <param name="type">Channel type</param>
+    /// <param name="parent">Channel parent ID</param>
+    /// <param name="topic">Channel topic</param>
+    /// <param name="bitrate">Voice channel bitrate</param>
+    /// <param name="userLimit">Voice channel user limit</param>
+    /// <param name="overwrites">Channel overwrites</param>
+    /// <param name="nsfw">Whether this channel should be marked as NSFW</param>
+    /// <param name="perUserRateLimit">Slow mode timeout for users.</param>
+    /// <param name="qualityMode">Voice channel video quality mode.</param>
+    /// <param name="position">Sorting position of the channel.</param>
+    /// <param name="reason">Reason this channel was created</param>
+    /// <param name="defaultAutoArchiveDuration">Default duration for newly created forum posts in the channel.</param>
+    /// <param name="defaultReactionEmoji">Default emoji used for reacting to forum posts.</param>
+    /// <param name="availableTags">Tags available for use by forum posts in the channel.</param>
+    /// <param name="defaultSortOrder">Default sorting order for forum posts in the channel.</param>
+    /// <exception cref="ArgumentException">The channel type must be a category, text, news, forum, voice or stage.</exception>
+    /// <returns>The newly created channel.</returns>
+    public Task<DiscordChannel> CreateGuildChannelAsync
+    (
+        ulong id,
+        string name,
+        ChannelType type,
+        ulong? parent,
+        Optional<string> topic,
+        int? bitrate,
+        int? userLimit,
+        IEnumerable<DiscordOverwriteBuilder> overwrites,
+        bool? nsfw,
+        Optional<int?> perUserRateLimit,
+        VideoQualityMode? qualityMode,
+        int? position,
+        string reason,
+        AutoArchiveDuration? defaultAutoArchiveDuration = null,
+        DefaultReaction? defaultReactionEmoji = null,
+        IEnumerable<DiscordForumTagBuilder>? availableTags = null,
+        DefaultSortOrder? defaultSortOrder = null
+    ) => type switch
```

Comment:
please put this in its own block, not an expression body

Reply:
The `.editorconfig` will cause a compiler warning, informing us to make it an expression. What is your proposed solution?

Reply:
change it to only prefer expressions on single lines?